clustering/ejb/WFLY-14359_Create_distributable-ejb_subsystem.adoc

```diff
@@ -0,0 +1,188 @@
+
+== Backwards Compatibility
+
+These changes affect backward compatability (c.f. legacy configuration of cache factories, passivation stores,
+client mappings registries, etc)
+
+=== Default Configuration
+
+Here is an example of the way the new subsystem will look, for the case of specifying a bean management provider:
+----
+<subsystem xmlns="urn:jboss:domain:distributable-ejb:1.0" default-bean-management="infinispan">
+    <infinispan-bean-management name="infinispan" cache-container="ejb" cache="dist" max-size="10"/>
+</subsystem>
+----
+and here is its corresponding integration into the ejb3 subsystem:
+----
+<subsystem xmlns="urn:jboss:domain:ejb3:10.0">
+     ...
+     <caches>
+        <simple-cache name="simple-cache"/>
+        <distributable-cache name="distributable-cache" bean-management="infinispan"/>
+    </caches>
+    ...
+ </subsystem>
+----
+Note that the distributable-cache element takes an optional bean-management attribute, which is a reference to
+the bean management provider in the new subsystem. A similar arrangment will be used for client mappings registries.
+When unspecified, the default-bean-management profile is used.
+
+As before, for each EJB SFSB, the @Cache annotation (resp. deployment descriptor configuration) may be used to specify
+the name of the desired cache factory defined in the ejb3 subsystem. If no @Cache annotation (resp. deployment descriptor
+configuration) is present, a suitable default cache will be used for the bean.
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+A new deployment configuration namespace, specified via jboss-all.xml or a separate distributable-ejb.xml, will be introduced
+to permit specifying default providers for bean management (resp. client-mappings management) on a per-deployment basis.
+This means that named, configured providers for bean management (resp. client-mappings management) may be specified in
+several places:
+
+* in the distributable-ejb subsystem, providing server-wide default provider values
+* in the distributable-ejb.xml file included with a deployment, providing deployment-scoped default provider values
+
+=== Interoperability
+
+== Implementation Plan
+
+As mentioned in the overview, the distributable-ejb subsystem will eventually support the provision of providers for four
+key areas of ejb3 subsystem functionality. However, this issue will be restricted to the following scope:
+
+* defining the distributable-ejb subsystem itself
+* defining a cache factory provider for stateful session beans and integrating that provider into the ejb3 subsystem
+* defining a client mappings registry provider for the EJB client and integrating that provider into the ejb3 subsystem
+
+The remaining two features will be completed in the separate RFEs mentioned above.
+
+== Security Considerations
+
+None
+
+== Test Plan
+
+The following areas of testing will be required:
+
+* integration of the EJB3 subsystem and the distributable-ejb subsystem
```

Comment:
Since this RFE will change the default configuration for both the basic and HA profiles, the integration between the ejb3 subsystem and the new distributable-ejb subsystem will be inherently validated by all existing EJB tests in the WF basic and clustering integration testsuites.  A new test will be added to validate that the legacy configuration (i.e.from WF26 or earlier, where the distributable-ejb subsystem is not present) still works.  This will reside in the clustering integration testsuite.

Reply:
Thanks for clarifying @pferraro.

@rachmatowicz - could you please incorporate the content that Paul shared here into the proposal?

Reply:
Done.